src/node.rs

```diff
@@ -14,161 +12,212 @@ use crate::style::*;
 
 type MeasureFunc = Box<Fn(Size<Number>) -> Result<Size<f32>>>;
 
-pub(crate) struct InternalNode {
-    pub(crate) style: Style,
-    pub(crate) parents: Vec<Weak<RefCell<InternalNode>>>,
-    pub(crate) children: Vec<Rc<RefCell<InternalNode>>>,
-    pub(crate) measure: Option<MeasureFunc>,
-    pub(crate) layout_cache: Option<Cache>,
-    pub(crate) is_dirty: bool,
+pub type NodeId = usize;
```

Comment:
Can we use newtype pattern here and export type as `Node`? I would like users of the library to think of this as a node instance rather than leaking implementation details. That way we could also implement methods on `Node` instead of implementing them on `Stretch` and passing in a node id. What are your thoughts?

Reply:
I moved to a newtype as you mentioned.

> That way we could also implement methods on Node instead of implementing them on Stretch and passing in a node id. What are your thoughts?

This would result in an API similar to https://github.com/saschagrunert/indextree#example-usage where we need to pass the corresponding `Stretch` instance in the call. I'm not having a real preference on this one. There would be option to support both (implement it on Stretch and on Node) tho.

Reply:
I agree. We can keep API as is.